test/system/members_test.rb

```diff
@@ -3,9 +3,25 @@
 require 'application_system_test_case'
 
 class MembersTest < ApplicationSystemTestCase
-  # test "visiting the index" do
-  #   visit members_url
-  #
-  #   assert_selector "h1", text: "Member"
-  # end
+  setup do
```

Comment:
I'm a little torn on the use of a system test here vs. a controller test. There are pros and cons of each, but a few things to think about:

* System tests are much slower than controller tests because they need to spin up a browser process
* There is an existing `MembersControllerTest`, which doesn't test the search results code path, and this test, which omits the "no search provided" code path. This makes it hard to keep the context of what is actually being tested.

Due to the runtime speed and context differentials, I'd recommend sticking to the controller tests for now. This is certainly up for discussion, but my personal feeling is that the _system tests_ should be treated a bit more like high-level integration tests and shouldn't necessarily be a replacement for what Rails refers to as a controller _functional test_.

I think Thoughtbot did a nice job of demonstrating some of these concepts in their post [How We Test Rails Applications](https://robots.thoughtbot.com/how-we-test-rails-applications). Also check out the official Rails [Guide to Testing Rails Applications](http://guides.rubyonrails.org/testing.html) for some context on Rails' usage expectations.

Reply:
Thanks @phlipper for the thought-provoking reply.  I *think* this quote from that Thoughtbot article is what you're advocating.
> When testing multiple paths through a controller is necessary, we favor using controller specs over feature specs, as they are faster to run and often easier to write.

I read the article as I was studying how to test, but it didn't resonate until now.  I've also been concerned that system tests are premature, in this case, because the view requirements are immature.